sdk/storage/inc/blobs/blob_client.hpp

```diff
@@ -130,6 +130,13 @@ namespace Azure { namespace Storage { namespace Blobs {
      */
     std::string GetUri() const { return m_blobUrl.ToString(); }
 
+    /**
+     * @brief Sets the blob's primary uri endpoint.
+     *
+     * @param uri the uri to be set as the blob's primary endpoint.
+     */
+    void SetUri(UrlBuilder uri) { m_blobUrl = std::move(uri); }
+
```

Comment:
nah, this makes this client non-thread-safe

Reply:
Removed and did it in another way.